### PR TITLE
hostap: support enterprise and DPP

### PR DESCRIPTION
**Pull request by fengming-ye**
Change cmake and kconfig for enterprise and DPP support.

### CODE REVIEW

**Comment by fengming-ye** [comment]
There might be a CI issue in Complicance check fail.
warning: undefined symbol MBEDTLS_MAC_CMAC_ENABLED:
but MBEDTLS_MAC_CMAC_ENABLED is defined in modules/mbedtls/Kconfig.tls-generic

**Review by krish2718 — COMMENTED** [review]
From a quick look changes LGTM, but can you please divide in to smaller commits with appropriate commit logs? 

E.g.

- NO_PBKDF2
- Removing blobs and why
- Adding CRC32
- Moving existing crypto
- Adding DPP
- Adding ALT crypto
...

**Comment by fengming-ye** [comment]
> From a quick look changes LGTM, but can you please divide in to smaller commits with appropriate commit logs?
> 
> E.g.
> 
> * NO_PBKDF2
> * Removing blobs and why
> * Adding CRC32
> * Moving existing crypto
> * Adding DPP
> * Adding ALT crypto
>   ...

updated pls check

**Comment by fengming-ye** [comment]
There might be a CI issue in compliance check fail
 Undefined Kconfig symbols:

 warning: undefined symbol MBEDTLS_MAC_CMAC_ENABLED:

But MBEDTLS_MAC_CMAC_ENABLED is defined in modules/mbedtls/Kconfig.tls-generic like other symbols.

**Comment by dleach02** [comment]
> MBEDTLS_MAC_CMAC_ENABLED

@fengming-ye, this changed in a PR that merged yesterday (things move fast) [#73267]. All the MBED_MAC_<algorithm>_ENABLED have changed to just MBED_<algorithm>

File: modules/hostap/Kconfig [hunk 1]
```diff
@@ -125,11 +127,37 @@ config WIFI_NM_WPA_SUPPLICANT_CRYPTO
...
 	select MBEDTLS_KEY_EXCHANGE_ALL_ENABLED
 
+config WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT
+	bool "Crypto Mbedtls alt support for WiFi"
+	select MBEDTLS
+	select MBEDTLS_CIPHER_MODE_CTR_ENABLED
+	select MBEDTLS_CIPHER_MODE_CBC_ENABLED
+	select MBEDTLS_ECP_C
+	select MBEDTLS_ECP_ALL_ENABLED
+	select MBEDTLS_MAC_CMAC_ENABLED
```
Comment (dleach02) [inline modules/hostap/Kconfig:137]: Change this to MBEDTLS_CMAC

**Comment by fengming-ye** [comment]
Hi @jukkar  @krish2718 
Here is kconfig and cmake changes for WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT
Pls also help review.

Hi @ceolin @d3zd3z @ithinuel 
Here is some kconfig adds for MBEDTLS_CONFIG macros.
Pls help review.

Thanks.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by dleach02 — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

**Pull request merged**